### PR TITLE
e2e: enrich phase matcher

### PR DESCRIPTION
**Pull request by fossedihelm**
Signed-off-by: fossedihelm <ffossemo@redhat.com>

**What this PR does / why we need it**:
Adds `kind` and the `name` of the object (if exist) when the phase matcher fails. 
Makes it easier to spot the failing object.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes [https://github.com/kubevirt/kubevirt/issues/7216](https://github.com/kubevirt/kubevirt/issues/7216)

**Special notes for your reviewer**:
Moved the `toUnstructured(actual interface{})` function from `owner.go` to `helper.go`

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/framework/matcher/phase.go [hunk 1]
```diff
@@ -68,31 +68,39 @@ func (p phaseMatcher) FailureMessage(actual interface{}) (message string) {
...
 		return fmt.Sprintf("object does not exist")
 	}
+	u, err := helper.ToUnstructured(actual)
+	if err != nil {
+		return err.Error()
+	}
 	expectedPhase := getExpectedPhase(p.expectedPhase)
 	if helper.IsSlice(actual) {
-		return fmt.Sprintf("expected phases to be in %v but got %v", expectedPhase, collectPhasesForPrinting(actual))
+		return fmt.Sprintf("%s %s expected phases to be in %v but got %v", u.GetKind(), u.GetName(), expectedPhase, collectPhasesForPrinting(actual))
```
Comment (xpivarc) [inline tests/framework/matcher/phase.go:77]: I would use `%s/%s` to match what kubectl uses for example.

**Review by xpivarc — COMMENTED** [review]
I think it may be good to exercise this with tests. WDYT?

File: tests/framework/matcher/phase.go (hunk 1 above)
Comment (fossedihelm) [reply tests/framework/matcher/phase.go:77]: Thanks, will fix, and add tests

**Comment by fossedihelm** [comment]
@xpivarc tests reveal some corner cases. Thanks. I changed the format of the `Sprintf` to `%s/%s`. Moreover, I add a nil check to the unstructured object and another one to the existence of the fields `Kind` and `Name`. I notice that in the `functest` the `Kind` field is always blank. Do you know the reason?

**Comment by xpivarc** [comment]
> @xpivarc tests reveal some corner cases. Thanks. I changed the format of the `Sprintf` to `%s/%s`. Moreover, I add a nil check to the unstructured object and another one to the existence of the fields `Kind` and `Name`. I notice that in the `functest` the `Kind` field is always blank. Do you know the reason?

Is it always empty? I think it should not be empty(populated by the client).

File: tests/framework/matcher/phase_test.go [hunk 2]
```diff
@@ -59,4 +93,33 @@ var _ = Describe("Matcher", func() {
 		table.Entry("cope with a non-stringable object as expected phase", nil, []*v1.Pod{runningPod}, false),
 		table.Entry("with expected phase not match the pod phase", "Succeeded", []*v1.Pod{runningPod}, false),
 	)
+
```
Comment (xpivarc) [inline tests/framework/matcher/phase_test.go:96]: Maybe add a case when only `name` is set

Comment (fossedihelm) [reply tests/framework/matcher/phase_test.go:96]: Sure. Thanks

**Review by xpivarc — APPROVED** [review]
Thank you once again!

File: tests/framework/matcher/phase_test.go [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+	"fmt"
+
 	. "github.com/onsi/ginkgo"
 	"github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
 	v1 "k8s.io/api/core/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v13 "kubevirt.io/api/core/v1"
+	"kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
```
Comment (EdDev) [inline tests/framework/matcher/phase_test.go:13]: It is odd to see CDI in e2e infrastructure from kubevirt/kubevirt. Even if it is in tests (which is odd by itself to see unit tests under the e2e tests folder).

I think it would have been nicer and cleaner to exercise this against kubevirt/kubevirt API.

Comment (xpivarc) [reply tests/framework/matcher/phase_test.go:13]: I will disagree. See https://github.com/kubevirt/kubevirt/blob/74a42a02f38f32299f55aae71f89d53b68344516/tests/storage/datavolume.go as an example where we also use `cdi`. Also, we have tide integration with datavolumes as we are embedding it to our api.

Comment (EdDev) [reply tests/framework/matcher/phase_test.go:13]: You have linked to storage sub-package, which is focused on storage.
Here we have a "framework" or library that is suppose to be independent and let others use it without also importing CDI stuff as a side-effect.

Comment (xpivarc) [reply tests/framework/matcher/phase_test.go:13]: I think we broke the assumption long age, see https://github.com/kubevirt/kubevirt/blob/74a42a02f38f32299f55aae71f89d53b68344516/tests/framework/matcher/getter.go#L13 (Or maybe I am missing something)

Comment (rmohr) [reply tests/framework/matcher/phase_test.go:13]: We never had any constraints like @EdDev mentioned. All good.

Comment (EdDev) [reply tests/framework/matcher/phase_test.go:13]: It is a good time to start having constraints on dependencies.
If you think it is ok to have such dependencies, I will be glad to hear a good reasoning for it.

Comment (xpivarc) [reply tests/framework/matcher/phase_test.go:13]: Let's discuss this in another place as I don't think it fits here.
